app/src/scripts/dashboard/dashboard.searchresults.jsx

```diff
@@ -32,14 +34,55 @@ class SearchResults extends React.Component {
     }
   }
 
+  render() {
+    let query = this.props.match.params.query || this.state.query
+    return (
+      <div className="route-wrapper">
+        <div className="fade-in inner-route clearfix">
+          <div className="dashboard-dataset-teasers datasets datasets-private">
+            <div className="admin clearfix">
+              <div className="row">
+                <div className="col-md-5">
+                  {query && this.state.results ? (
+                    <React.Fragment>
+                      <h2>Search Results</h2>
+                      <span className="sub-title">
+                        showing {this.state.results.length} results for: {query}
+                      </span>
+                    </React.Fragment>
+                  ) : null}
+                </div>
+                <div className="col-md-7">
+                  <Search />
+                </div>
+              </div>
+              <div className="panel-group">
+                {this.state.loading ? (
+                  <Spinner active={true} />
+                ) : query ? (
+                  this._results(this.state.results)
+                ) : (
```

Comment:
We should avoid nested ternary expressions, this is hard to read. Better to make a function with if statements or a small component if the logic is complex enough.

Reply:
- [x] refactored to be nested conditionals within templating methods